extension/test/data_test.ts

```diff
@@ -69,4 +69,209 @@ describe('data.ts', function () {
+        data: [],
+        hasMore: false,
+        title: '',
+        index: 0,
+        matchLen: 0,
+      };
+
+      const kanjiDictEntry = {
+        ...unimportantProperties,
+        kanji: '<kanji-entry>',
+        onkun: '<onkun-entry-1>、 <onkun-entry-2>、 <onkun-entry-3>',
+        nanori: '<nanori-entry>',
+        bushumei: '<bushumei-entry>',
+        misc: {
+          U: '<U-misc-entry>',
+          B: '<B-misc-entry>',
+          G: '<G-misc-entry>',
+          S: '<S-misc-entry>',
+          F: '<F-misc-entry>',
+          N: '<N-misc-entry>',
+          V: '<V-misc-entry>',
+          H: '<H-misc-entry>',
+          DK: '<DK-misc-entry>',
+          DL: '<DL-misc-entry>',
+          L: '<L-misc-entry>',
+          DN: '<DN-misc-entry>',
+          E: '<E-misc-entry>',
+          IN: '<IN-misc-entry>',
+          P: '<P-misc-entry>',
+          I: '<I-misc-entry>',
+          Y: '<Y-misc-entry>',
+        },
+        eigo: '<eigo-entry-1>; <eigo-entry-2>; <eigo-entry-3>',
+      };
+
+      it('returns 1 kanji entry formatted as text with newlines separating kanji, eigo, onkun, nanori, bushumei, and misc lines; misc name and entry separated by a tab; ending in newline', function () {
+        const singleKanjiDataEntryAsText = rcxDict.makeText(
+          kanjiDictEntry,
+          /* max= */ 1
+        );
+
+        expect(singleKanjiDataEntryAsText).to.equal(
+          '<kanji-entry>\n<eigo-entry-1>; <eigo-entry-2>; <eigo-entry-3>\n<onkun-entry-1>、 <onkun-entry-2>、 <onkun-entry-3>\n名乗り\t<nanori-entry>\n部首名\t<bushumei-entry>\nHalpern\t<H-misc-entry>\nHeisig 5th Edition\t<L-misc-entry>\nHeisig 6th Edition\t<DN-misc-entry>\nHenshall\t<E-misc-entry>\nKanji Learners Dictionary\t<DK-misc-entry>\nKanji Learners Dictionary 2nd Edition\t<DL-misc-entry>\nNelson\t<N-misc-entry>\nNew Nelson\t<V-misc-entry>\nPinYin\t<Y-misc-entry>\nSkip Pattern\t<P-misc-entry>\nTuttle Kanji & Kana\t<IN-misc-entry>\nTuttle Kanji Dictionary\t<I-misc-entry>\nUnicode\t<U-misc-entry>\n'
+        );
+      });
+
+      it('even with large max, returns a single kanji data entry as text', function () {
+        const singleKanjiDataEntryAsText = rcxDict.makeText(
+          kanjiDictEntry,
+          /* max= */ 1000
+        );
+
+        expect(singleKanjiDataEntryAsText).to.equal(
+          '<kanji-entry>\n<eigo-entry-1>; <eigo-entry-2>; <eigo-entry-3>\n<onkun-entry-1>、 <onkun-entry-2>、 <onkun-entry-3>\n名乗り\t<nanori-entry>\n部首名\t<bushumei-entry>\nHalpern\t<H-misc-entry>\nHeisig 5th Edition\t<L-misc-entry>\nHeisig 6th Edition\t<DN-misc-entry>\nHenshall\t<E-misc-entry>\nKanji Learners Dictionary\t<DK-misc-entry>\nKanji Learners Dictionary 2nd Edition\t<DL-misc-entry>\nNelson\t<N-misc-entry>\nNew Nelson\t<V-misc-entry>\nPinYin\t<Y-misc-entry>\nSkip Pattern\t<P-misc-entry>\nTuttle Kanji & Kana\t<IN-misc-entry>\nTuttle Kanji Dictionary\t<I-misc-entry>\nUnicode\t<U-misc-entry>\n'
+        );
+      });
+
+      describe('empty fields', function () {
```

Comment:
maybe `with empty fields` for consistency; it would probably make sense either way in the final output...